algebras/algebra/src/test/scala/endpoints4s/algebra/server/EndpointsTestSuite.scala

```diff
@@ -460,6 +460,35 @@ trait EndpointsTestSuite[T <: endpoints4s.algebra.EndpointsTestApi] extends Serv
       }
     }
 
+    "reject requests with missing headers" in {
```

Comment:
Wouldn't it be better to split the cases in separate tests? In case one fails the whole test fails